### PR TITLE
Logging code cleanup related to Nomad auto-discovery

### LINKED ISSUE #26456: Nomad auto-discover uses Debugf incorrectly
The nomad auto-discover provider invokes the `logp.Logger`'s `Debugf` with an incorrect set of arguments. I noticed two of them, but I recommend doing a quick audit of the rest of the usages in the package to make sure they are all fixed.

```
{
  "level": "debug",
  "timestamp": "2021-06-23T21:48:38.451Z",
  "logger": "nomad",
  "caller": "nomad/nomad.go:283",
  "message": "nomad%!(EXTRA string=Generated builder event %+v, bus.Event=map[....REDACTED"
}
{
  "level": "debug",
  "timestamp": "2021-06-23T21:48:38.452Z",
  "logger": "nomad",
  "caller": "nomad/nomad.go:228",
  "message": "nomad%!(EXTRA string=Publishing event: %+v, bus.Event=map[....REDACTED"
}
```

Version: `filebeat version 8.0.0 (arm64), libbeat 8.0.0 [0ae157ed7c4341035371655cbf426e2f071bc178 built 2021-06-23 21:01:19 +0000 UTC]`

### PR DESCRIPTION
**Pull request by andrewkroh**
## What does this PR do?

This does some minor clean up of the logging code and docs related to Nomad. I removed some statements that were providing duplicate information and switched a few over to using the structured logger to improve log readability.

A log message about `enabled` not being parsed was always being logged despite no actual error.

The doc examples were trying to start a Filebeat input with `type: nomad`.

Fixed some typos in code comments.

## Why is it important?

This make debugging a little more enjoyable 😆 .

## Checklist

- [x] My code follows the style guidelines of this project
~- [ ] I have commented my code, particularly in hard-to-understand areas~
- [x] I have made corresponding changes to the documentation
~- [ ] I have made corresponding change to the default configuration files~
~- [ ] I have added tests that prove my fix is effective or that my feature works~
~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~

## Related issues

- Closes #26456

### CODE REVIEW

**Comment by andrewkroh** [comment]
run tests

File: filebeat/docs/autodiscover-hints.asciidoc [hunk 1]
```diff
@@ -278,31 +281,81 @@ filebeat.autodiscover:
...
 filebeat.autodiscover:
   providers:
     - type: nomad
       hints.enabled: true
-      hints.default_config.enabled: false
+      hints.default_config:
+        enabled: false
+        type: log
+        paths:
+          - /opt/nomad/alloc/${data.nomad.allocation.id}/alloc/logs/${data.nomad.task.name}.*
```
Comment (wjordan) [inline filebeat/docs/autodiscover-hints.asciidoc:302]: I have a semi-related note about this path template (sorry if slightly off-topic)- when testing out this feature I noticed that the `task.name` data was only being published on the `meta.nomad` field of the event, not the `nomad` field used in this example. Here's the relevant code: https://github.com/elastic/beats/blob/076e0a693c18617454388c6759c4010e821c0f59/x-pack/libbeat/autodiscover/providers/nomad/nomad.go#L197-L210
In my tests, `${data.nomad.task.name}` gave an invalid template reference error, but `${data.meta.nomad.task.name}` resolved properly. Does this sound correct and if so, should all the documentation examples be updated to reflect this working usage? Or does a separate PR to fix the existing examples make more sense?

**Comment by andrewkroh** [comment]
Note to self, two other things to fix:

- Docs show `secretID` as config value but it's `secret_id`.

- Docs are missing info about ACL requirements. Seems like filebeat needs an ACL token with these capabilities.

```
node {
  policy = "read"
}
agent {
  policy = "read"
}
```

File: filebeat/docs/autodiscover-hints.asciidoc (hunk 1 above)
Comment (andrewkroh) [reply filebeat/docs/autodiscover-hints.asciidoc:302]: I noted that there was a lot of duplicate information in the autodiscover events that get published by that code. Here's a sample document.

Comment (wjordan) [reply filebeat/docs/autodiscover-hints.asciidoc:302]: > Can you open a separate issue please?

Opened #26538.

> But I wasn't sure what the expected behavior was.

It turns out there was some discussion around the need for this duplicate information in the original PR- see https://github.com/elastic/beats/pull/14954#discussion_r398699442.

Comment (andrewkroh) [reply filebeat/docs/autodiscover-hints.asciidoc:302]: Great find. So based on that comment it sounds like we need to keep both `meta` and `nomad`. But we can make sure all the documented keys are present under the `nomad.`

**Comment by andrewkroh** [comment]
run tests

File: filebeat/docs/autodiscover-hints.asciidoc (hunk 1 above)
Comment (jsoriano) [reply filebeat/docs/autodiscover-hints.asciidoc:302]: Good catch!

I think only the `nomad` fields need to be available for config templates. The meta fields are used for enrichment, but shouldn't be necessarily available for templates. We probably messed something up in the original PR, I remember going back and forth on this :grimacing: 

We also have to double-check that hints based configs have the same fields available.

**Review by jsoriano — APPROVED** [review]
Thanks! Glad to see some activity around nomad :slightly_smiling_face:

**Pull request merged**